modules/core/src/main/java/org/mycontroller/standalone/db/dao/GatewayDaoImpl.java

```diff
@@ -77,7 +77,7 @@ public GatewayDaoImpl(ConnectionSource connectionSource) throws SQLException {
             return gateways;
         } catch (SQLException ex) {
             _logger.error("unable to get all gateways:[type:{}, NetworkType:{}, Enabled:{}]",
-                    gatewayType.getText(), networkType, enabled, ex);
+                    gatewayType != null ? gatewayType.getText() : "no gateway", networkType, enabled, ex);
```

Comment:
@Fraid we can pass `null` as the type when we do not have gatewayType object. "no gateway" will confuse the user.